runtime/metricsview/astsql.go

```diff
@@ -217,7 +218,24 @@ func (b *sqlBuilder) writeSelect(n *SelectNode) error {
 	return nil
 }
 
-func (b *sqlBuilder) writeJoin(joinType JoinType, baseSelect, joinSelect *SelectNode) error {
+func (a *AST) interval() (string, error) {
+	if a.query.TimeRange == nil {
+		return "", fmt.Errorf("no time range for the offset")
+	}
+	if a.query.TimeRange.Start.IsZero() {
+		return "", fmt.Errorf("no start time for the offset")
+	}
+	if a.query.ComparisonTimeRange == nil {
+		return "", fmt.Errorf("no comparison time range for the offset")
+	}
+	if a.query.ComparisonTimeRange.Start.IsZero() {
+		return "", fmt.Errorf("no start time for the comparison time range")
+	}
+	diff := a.query.ComparisonTimeRange.Start.Sub(a.query.TimeRange.Start)
+	return fmt.Sprint(diff.Milliseconds()), nil
```

Comment:
Unfortunately it's not sufficient to do millisecond delta comparisons. For example, if the time granularity is monthly, the delta will be different for each month, so you need to use `INTERVAL 1 MONTH` in SQL instead of microseconds.

It may need to scan the dimensions for the biggest granularity used in `TimeFloor` and use that to determine the interval type. Also, see the issue description for some other thoughts: https://github.com/rilldata/rill/issues/5361

Reply:
Done.